pkg/flowaggregator/s3uploader/s3uploader.go

```diff
@@ -136,7 +136,7 @@ func NewS3UploadProcess(input S3Input, clusterUUID string) (*S3UploadProcess, er
 
 	buf := &bytes.Buffer{}
 	// #nosec G404: random number generator not used for security purposes
-	nameRand := rand.New(rand.NewSource(time.Now().UnixNano()))
+	nameRand := rand.New(rand.NewPCG(uint64(time.Now().UnixNano()), 0)) // Updated initialization
```

Comment:
we should use the main random generator provided by the package here (see my comment in the issue, where I answered your question). The main generator is no longer deterministic.